### PR TITLE
Implement CDC iterator

### PR DESCRIPTION
**Pull request by alyakimenko**
### Description

This PR seeks to implement CDC iterator.

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/conduitio-labs/conduit-connector-hubspot/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

**Review by voscob — APPROVED** [review]
👍

File: source/iterator/cdc.go [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+				c.errC <- fmt.Errorf("load records: %w", err)
+			}
+		}
+	}
+}
+
+// loadRecords loads HubSpot items using timestamp-based filtering and search endpoints.
+// The method tries to retrieve items filtering them by updatedAfter or lastmodifieddate.
+func (c *CDC) loadRecords(ctx context.Context) error {
+	// add a millisecond here in order to skip the processed item
```
Comment (maksenius) [inline source/iterator/cdc.go:119]: it it possible to have some items with the same timestamp?

File: source/iterator/cdc.go [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+	if searchResource, ok := hubspot.SearchResources[c.resource]; ok {
+		return c.fetchSearchBasedItems(ctx, searchResource, updatedAfter)
+	}
+
+	return nil
+}
+
+// fetchTimestampBasedItems fetches items by filtering them by updatedAt field.
+func (c *CDC) fetchTimestampBasedItems(
+	ctx context.Context, resource hubspot.TimestampResource, updatedAfter time.Time,
```
Comment (maksenius) [inline source/iterator/cdc.go:144]: could you please change this style `ctx context.Context, resource hubspot.TimestampResource, updatedAfter time.Time,`
to it:
```
ctx context.Context,
resource hubspot.TimestampResource, 
 updatedAfter time.Time,
```

Comment (maksenius) [reply source/iterator/cdc.go:144]: in other places please do the same

File: source/iterator/cdc.go [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+		// as we sort items by their updatedAt values.
+		Timestamp: &itemUpdatedAt,
+	}
+
+	sdkPosition, err := c.position.MarshalSDKPosition()
+	if err != nil {
+		return fmt.Errorf("marshal sdk position: %w", err)
+	}
+
+	if itemCreatedAt.After(updatedAfter) {
```
Comment (maksenius) [inline source/iterator/cdc.go:216]: maybe we can add comment to explain why we do it

File: source/iterator/cdc.go (hunk 2 above)
Comment (alyakimenko) [reply source/iterator/cdc.go]: Okay, I've changed the style.

Comment (maksenius) [reply source/iterator/cdc.go]: thank you!

File: source/iterator/cdc.go (hunk 3 above)
Comment (alyakimenko) [reply source/iterator/cdc.go:225]: I've added the comment explaining this behavior.

File: source/iterator/cdc.go (hunk 1 above)
Comment (alyakimenko) [reply source/iterator/cdc.go:119]: In case two records are created simultaneously (up to milliseconds).

I'll check what else we can do with this.

Comment (maksenius) [reply source/iterator/cdc.go:119]: I mean batch creating, if difference is milliseconds it is ok

Comment (alyakimenko) [reply source/iterator/cdc.go:119]: @maksenius I've tried bulk import supported by HubSpot, and it looks like the createdAt and updatedAt fields are different for each element. So, yeah, even with bulk import, the timestamps are different.

Comment (maksenius) [reply source/iterator/cdc.go:119]: cool

**Review by maksenius — APPROVED** [review]

**Pull request merged**